Add TIBCO migration docs (#71)
wso2/docs-bi: en/docs/learn/tibco-migration-assistant.md

# TIBCO BusinessWorks Migration
This guide explains how to use the [migrate-tibco](https://central.ballerina.io/wso2/tool_migrate_tibco/latest) tool to convert
[TIBCO BusinessWorks](https://docs.tibco.com/products/tibco-activematrix-businessworks) integrations into Ballerina packages compatible with the [WSO2 Ballerina Integrator](https://wso2.com/integrator/ballerina-integrator).
## Tool Overview

> he tool accepts either a BusinessWorks project directory or a standalone `bwp` file as input and generates an equivalent Ballerina package that can be opened in WSO2 Ballerina Integrator.

Reviewer: ```suggestion
The tool accepts either a BusinessWorks project directory or a standalone `bwp` file as input and generates an equivalent Ballerina package that can be opened in WSO2 Ballerina Integrator.
```

## Installation

Execute the command below to pull the `migrate-tibco` tool from Ballerina Central
```bash
$ bal tool pull migrate-tibco
```

## Usage

### Command Syntax

```bash
$ bal migrate-tibco <source-project-directory-or-file> [-o|--out <output-directory>]
```

### Parameters

> - **source-project-directory-or-file** - Required. The TIBCO BusinessWorks project directory or `bwp` file to migrate.

Reviewer: ```suggestion
- **source-project-directory-or-file** - Required. The TIBCO BusinessWorks project directory or `bwp` file needs to be migrated
```

> - **-o or --out** - *Optional*. The directory where the new Ballerina package will be created. If the directory does not exist tool will create it for you. If not provided,

Reviewer: ```suggestion
- **-o or --out** - *Optional*. The directory where the new Ballerina package will be created. If the directory does not exist, the tool will create it for you. If not provided,
```

  - If source-project-directory-or-file is a directory it will create new directory named ${source-project-directory-or-file}_converted in the root of source-project-directory-or-file

>   - if source-project-directory-or-file is a file it will create a new directory named ${root}_converted in the parent of the root directory where root is the directory containing the given file.

Reviewer: ```suggestion
  - if source-project-directory-or-file is a file, it will create a new directory named ${root}_converted in the parent of the root directory where root is the directory containing the given file.
```

## Example

### Step 1

> 1. Pull `migrate-tibco` tool for Ballerina Central using fallowing command.

Reviewer: ```suggestion
1. Pull `migrate-tibco` tool for Ballerina Central using following command.
```

```bash
$ bal tool pull migrate-tibco
```

### Step 2

> 1. Create the fallowing `helloworld.bwp` file.

Reviewer: ```suggestion
1. Create the following `helloworld.bwp` file.
```

                                <tibex:inputBindings>
                                    <tibex:partBinding
                                    expression="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;xsl:stylesheet xmlns:xsl=&quot;http://www.w3.org/1999/XSL/Transform&quot; xmlns:tns=&quot;http://xmlns.example.com&quot; version=&quot;2.0&quot;&gt;
    &lt;xsl:template name=&quot;getOut-input&quot; match=&quot;/&quot;&gt;
        &lt;response&gt; helloworld &lt;/response&gt;
    &lt;/xsl:template&gt;
&lt;/xsl:stylesheet&gt;" expressionLanguage="urn:oasis:names:tc:wsbpel:2.0:sublang:xslt1.0"/>
                                </tibex:inputBindings>
                            </bpws:reply>
                        </bpws:flow>
                    </bpws:scope>
                </bpws:onMessage>
            </bpws:pick>
        </bpws:flow>
    </bpws:scope>
</bpws:process>
```
2. Execute `bal migrate-tibco <path to helloworld.bwp> -o converted`. This will create the `converted` directory and create a Ballerina package inside it.

### Step 3
1. Open VS Code inside the `converted` directory
```bash
$ code ./test

> ```

Reviewer: indent the code block

> 2. Click Ballerina Integrator button in the left side bar to open Ballerina package in Ballerina Integrator.

Reviewer: ```suggestion
2. Click the Ballerina Integrator icon on the left side bar to open the Ballerina package in Ballerina Integrator.
```

## Output

### Migration summary

> - When you run the tool it will log the number of activities it detected for each process along with the number of activities it failed to convert, if any.

Reviewer: ```suggestion
- When you run the tool, it will log the number of activities it detected for each process along with the number of activities it failed to convert, if any.
```

### Unhandled activities

- If the tool encounters any activity which it does not know how to convert it will generate a placeholder "unhandled" function with a comment containing the relevant part of the `bwp` file.

> ```

Reviewer: ```suggestion
```ballerina
```
